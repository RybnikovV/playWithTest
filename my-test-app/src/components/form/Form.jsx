import { useState } from 'react'

export const Form = () => {
  const [showLastName, setShowLastName] = useState(true);
  const [nameValue, setNameValue] = useState('')

  return (
    <>
      <input 
        value={nameValue}
        placeholder='Введите имя' 
        type='text'
        data-testid='inputTest'
        onChange={(e) => setNameValue(e.target.value)}
        >
        </input>
      {
        showLastName && <input 
          placeholder='Введите отчество'
          data-testid="inputElement"
          type="text" />
      }
      <button 
        onClick={() => setShowLastName(!showLastName)}
        data-testid="btnElement">
        {
          showLastName ? "У меня нет отчества" : "Ввести отчество"
        }
      </button>
      <button>Отправить</button>
    </>
  )
}