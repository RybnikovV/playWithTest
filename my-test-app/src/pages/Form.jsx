import { Form } from '../components/form/Form';

export const FormPage = () => {
  return (
    <div data-testid="form-page">
      <Form/>
    </div>
  )
};