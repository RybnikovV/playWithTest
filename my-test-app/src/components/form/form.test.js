import { render, screen, fireEvent} from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Form } from './Form';
import userEvent from '@testing-library/user-event';

describe('Проверка Form', () => {
  // test('renders form component', () => {
  //   render(<Form/>);
  //   const formInput = screen.getByPlaceholderText(/введите имя/i);
  //   const formButton = screen.getByRole('button');
  //   expect(formInput).toBeInTheDocument();
  //   expect(formButton).toBeInTheDocument();
  //   expect(formInput).toMatchSnapshot();
  //   // screen.debug();
  // });

  test('test click event', () => {
    act(() => {
      render(<Form/>);
    });
    const clickBtn = screen.queryByTestId('btnElement');
    const toggleInput = screen.queryByTestId('inputElement');
    expect(clickBtn).toBeInTheDocument();
    expect(toggleInput).toBeInTheDocument();
    expect(clickBtn).toHaveTextContent('У меня нет отчества');
    fireEvent.click(clickBtn);
    expect(screen.queryByTestId('inputElement')).toBeNull;
    expect(clickBtn).toBeInTheDocument();
    expect(clickBtn).toHaveTextContent('Ввести отчество');
  });

  test('test input event', () => {
    act(() => {
      render(<Form/>);
    });
    const input = screen.getByPlaceholderText(/введите имя/i);
    expect(input).toContainHTML('');
    expect(input).toBeInTheDocument();
    // fireEvent.input(input, {
    //   target: {value: '123'}
    // });
    act(() => {
      userEvent.type(input, '123');
    });
    expect(screen.getByTestId('inputTest')).toContainHTML('123');
  })
});