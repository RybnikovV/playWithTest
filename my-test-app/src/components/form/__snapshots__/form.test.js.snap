// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form component 1`] = `
<input
  placeholder="Введите имя"
  type="text"
/>
`;
